deploy-board/deploy_board/webapp/group_view.py

```diff
@@ -959,30 +959,39 @@ def add_instance(request, group_name):
     subnet = None
     placement_group = None
     asg_status = params['asgStatus']
-    launch_in_asg = True
-    if 'subnet' in params:
-        subnet = params['subnet']
-        if asg_status == 'UNKNOWN':
-            launch_in_asg = False
-
-        elif 'customSubnet' in params:
-            launch_in_asg = False
-            # custom subnet is selected for launching hosts. Only then make a check for Placement Group
-            if 'customPlacementGroup' in params and 'placementGroup' in params:
+    launch_in_asg = False
+
+    try:
+        # Configure subnet to launch hosts
+        if 'subnet' in params:
+            subnet = params['subnet']
+            # check if custom subnet is specified and custom placement group is in it.
+            if 'customSubnet' in params and 'customPlacementGroup' in params and 'placementGroup' in params:
                 # The check box is ticked and placement group is entered.
                 placement_group = params['placementGroup']
 
-    try:
-        if not launch_in_asg:
-            # Launch static hosts
+        # Check if asg is enabled. Then launch in asg
+        if str(asg_status).upper() == "ENABLED":
```

Comment:
If user set a custom subnet and placementgroup, even we have ASG enabled, I think we should still launch outside of the ASG.